# Experiment a1a8adc2d662a407af5840eabcf81aee34e4fefcd6b63ab915a9bf6c61000b24
# Base d9f0715b066f648c88e24b34486e3c4946234bf5
diff --git a/src/move_obj.cpp b/src/move_obj.cpp
index ca59188b2..aee882981 100644
--- a/src/move_obj.cpp
+++ b/src/move_obj.cpp
@@ -399,7 +399,8 @@ float RGE_Moving_Object::angle_to_object(RGE_Static_Object* target, float tx, fl
             ax = tx;
         }
 
-        float a = (float)atan2((double)(ay - world_y), (double)(ax - world_x));
+        float x_delta = ax - world_x;
+        float a = atan2(ay - world_y, x_delta);
         if (a < DAT_00570b48)
             a -= DAT_00570b4c;
         a -= angle;
